### ISSUE TITLE
`defineAuth` not enabling Hosted UI when passing social providers or OIDC

### ISSUE THREAD

**Issue by nadetastic** [aws-amplify/amplify-backend#734]
### Environment information

```plain text
System:
    OS: macOS 13.5.1
    CPU: (8) x64 Intel(R) Core(TM) i5-8257U CPU @ 1.40GHz
    Memory: 32.31 MB / 8.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.17.0 - ~/.nvm/versions/node/v18.17.0/bin/node
    Yarn: 1.22.5 - /usr/local/bin/yarn
    npm: 9.6.7 - ~/.nvm/versions/node/v18.17.0/bin/npm
    pnpm: 7.5.0 - /usr/local/bin/pnpm
    bun: Not Found
    Watchman: 2023.10.09.00 - /usr/local/bin/watchman
  npmPackages:
    @aws-amplify/backend: ^0.5.5 => 0.5.5 
    @aws-amplify/backend-cli: ^0.9.1 => 0.9.1 
    aws-amplify: ^6.0.5-unstable.b877493.0 => 6.0.5-unstable.b877493.0+b877493 
    aws-cdk: Not Found
    aws-cdk-lib: Not Found
    typescript: ^5.3.2 => 5.3.2
```

### Description

Attempting to setup OIDC (or social sign in) the Hosted UI isn't setup which doesn't allow the client/front end to successfully call `signInWithRedirect()` - https://docs.amplify.aws/react/build-a-backend/auth/add-social-provider/#set-up-your-frontend

```ts
import { defineAuth, secret } from "@aws-amplify/backend";

export const auth = defineAuth({
  loginWith: {
    email: true,
    externalProviders: {
      google: {
        clientId: secret("GOOGLE_CLIENT_ID"),
        clientSecret: secret("GOOGLE_CLIENT_SECRET"),
      },
      oidc: {
        clientId: secret("MYOIDC_CLIENT_ID"),
        clientSecret: secret("MYOIDC_CLIENT_SECRET"),
        issuerUrl: "https://example.com",
        name: "MYOIDC",
      },
    },
  },
});

**Comment by nadetastic** [comment]
Work around for this is to override the auth resources

```js
mybackend.resources.auth.resources.userPool.addDomain("cognito-domain", {
  cognitoDomain: {
    domainPrefix: "your-domain-prefix",
  },
});
```

**Comment by webjay** [comment]
@nadetastic Are you getting the error: `OAuthNotConfigureException: oauth param not configured.`?

I tried your hack, but it is not working for me.

```JavaScript
backend.auth.resources.userPool.addDomain('cognito-domain', {
  cognitoDomain: {
    domainPrefix: 'your-domain-prefix',
  },
});
```

- @aws-amplify/backend: 0.7.0
- aws-amplify: 6.0.9

**Comment by CojocaruDr** [comment]
Same issue as @webjay here. I was comparing the ``aws-exports.js`` generated by my gen1 vs gen2, and it certainly is missing that ``oauth`` key in gen2. The workaround doesn't seem to have anything to do with this though (doesn't work for me either).

I have reverted my changes and using gen1 now because of this reason, but I was using the same versions as @webjay. I think I still have a branch with the code somewhere so I can provide more info if needed.

**Comment by fossamagna** [comment]
This issue is fixed after [@aws-amplify/backend@0.8.0](https://github.com/aws-amplify/amplify-backend/releases/tag/%40aws-amplify%2Fbackend%400.8.0).

1. Add `callbackUrls`, `logoutUrls` and `domainPrefix` as following:
   ```ts
   import { defineAuth, secret } from ["@aws-amplify/backend"](https://www.npmjs.com/package/@aws-amplify/backend);

   export const auth = defineAuth({
     loginWith: {
       email: true,
       externalProviders: {
         google: {
           clientId: secret("GOOGLE_CLIENT_ID"),
           clientSecret: secret("GOOGLE_CLIENT_SECRET"),
         },
         oidc: {
           clientId: secret("MYOIDC_CLIENT_ID"),
           clientSecret: secret("MYOIDC_CLIENT_SECRET"),
           issuerUrl: "https://example.com",
           name: "MYOIDC",
         },
         callbackUrls: ['http://callback.com'],
         logoutUrls: ['http://logout.com'],
         domainPrefix: 'your-prefix',
       },
     },
    });
   ```
2. Deploy your project(ex: `git push` or `npx amplify sandbox`).
3. Run `npx amplify generate config`
4. Generated `amplifyconfiguration.json` included `oauth` key.

**Comment by ykethan** [comment]
Closing the issue as the issue has been fixed.
